### DISCUSSION TITLE
support monitoring by executing queries from the existing ES storage?

### DISCUSSION THREAD

**Discussion by leizhiyuan** [apache/hertzbeat#1729] [Ideas]
假设所有的数据都上报到了es/别的存储中存储，可以通过扩展hertzbeat的自定义采集协议，通过执行查询语法，来做监控和告警吗？ 

Assuming all the data has been reported to ES/another storage for storage, is it possible to use a customized collection protocol to do monitoring and alerting by extending Hertzbeat?

**Comment by tomsun28** [comment]
hi, in theory, it can be done. We can implement the es-based query protocol like other protocols and design the structure of the protocol.
